### PR TITLE
fix: c2 file upload bug workaround (FPLA-1219)

### PR DESCRIPTION
**Pull request by pd-dudkiewicz**
### Change description ###

"Replace existing file" was displayed on file upload after clicking previous and continue on c2 upload journey.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsMidEventControllerTest.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 @ActiveProfiles("integration-test")
 @WebMvcTest(UploadC2DocumentsController.class)
 @OverrideAutoConfiguration(enabled = true)
+@SuppressWarnings("unchecked")
```
Comment (swalker125) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsMidEventControllerTest.java]: I don't like this

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsMidEventControllerTest.java [hunk 2]
```diff
@@ -72,9 +73,9 @@ void shouldRemoveTemporaryC2DocumentForEmptyUrl() {
 
         AboutToStartOrSubmitCallbackResponse response = postMidEvent(CaseDetails.builder()
             .data(Map.of("temporaryC2Document", Map.of("document", Map.of())))
-            .build());
+            .build(), "get-fee");
 
-        assertThat(response.getData()).doesNotContainKey("temporaryC2Document");
+        assertThat((Map<String, Object>) response.getData().get("temporaryC2Document")).doesNotContainKey("document");
```
Comment (swalker125) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsMidEventControllerTest.java]: can we do assertThat(response.getData()).extracting("temporyC2Document")... here?

Comment (pd-dudkiewicz) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsMidEventControllerTest.java]: nice :+1: removed warning suppression thanks to that

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java [hunk 3]
```diff
@@ -53,19 +52,17 @@
     private final FeatureToggleService featureToggleService;
     private final PbaNumberService pbaNumberService;
 
-    @PostMapping("/mid-event")
+    @PostMapping("/get-fee/mid-event")
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java]: method name can be updated

**Pull request merged**